pkgs/top-level/all-packages.nix

```diff
@@ -5135,7 +5135,8 @@ with pkgs;
   eschalot = callPackage ../tools/security/eschalot { };
 
   espanso = callPackage ../applications/office/espanso {
-    inherit (darwin.apple_sdk.frameworks) AppKit Cocoa Foundation IOKit Kernel AVFoundation Carbon QTKit AVKit WebKit;
+    stdenv = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
```

Comment:
This can be simplified to just replacing `callPackage` with `darwin.apple_sdk_11.0.callPackage`

According to the documentation `doc/stdenv/platform-notes.chapter.md`:

> x86_64-darwin uses the 10.12 SDK by default, but some software is not compatible with that version of the SDK. In that case,
  the 11.0 SDK used by aarch64-darwin is available for use on x86_64-darwin. To use it, reference `apple_sdk_11_0` instead of
  `apple_sdk` in your derivation and use `pkgs.darwin.apple_sdk_11_0.callPackage` instead of `pkgs.callPackage`. On Linux, this will
  have the same effect as `pkgs.callPackage`, so you can use `pkgs.darwin.apple_sdk_11_0.callPackage` regardless of platform.

Nevertheless, this does not fix the build on darwin-x86_64. You seem to have quite a distinct problem. I have found a relatively similar issue here https://github.com/zed-industries/zed/issues/7036 , so it might be related to the clang version. Not sure why this happens on only a few packages, though, as I've upgraded a few packages to SDK11 with no issues.

Reply:
Removed the explicit sdk selection for now since it doesn't fully fix the build. I think this should be done as a follow-up and once a real solution is known.